src/librustc_mir/transform/inline.rs

```diff
@@ -395,16 +395,25 @@ impl Inliner<'tcx> {
                 let mut local_map = IndexVec::with_capacity(callee_body.local_decls.len());
                 let mut scope_map = IndexVec::with_capacity(callee_body.source_scopes.len());
 
-                for mut scope in callee_body.source_scopes.iter().cloned() {
+                for (callee_idx, scope) in callee_body.source_scopes.iter_enumerated() {
```

Comment:
cc @Centril @wesleywiser this PR may in fact be fixing the inliner ICE encountered during stage2 building